### ISSUE TITLE
Exclude parentheses from function parameter highlights

### ISSUE THREAD

**Issue by mariusschulz** [isagalaev/highlight.js#803]
Take the following piece of JavaScript code as an example:

```js
var add = function(a, b) {
    return a + b;
};
```

Here's how the above snippet is highlighted:

```html
<code class=" hljs javascript"><span class="hljs-keyword">var</span> add = <span class="hljs-function"><span class="hljs-keyword">function</span><span class="hljs-params">(a, b)</span> </span>{
    <span class="hljs-keyword">return</span> a + b;
};
</code>
```

Notice how parameters are marked up:

```html
<span class="hljs-params">(a, b)</span>
```

Wouldn't it be nicer if the parentheses weren't included in the `<span>`? All text editors I use (Sublime Text, Brackets, Visual Studio, …) treat the highlighting this way, so it seems weird to me to see the parentheses colored as well (but not the curly braces) when I style `.hljs-params` with a different color.

**Comment by isagalaev** [comment]
I personally wouldn't care either way but by your reasoning it does seem like a good idea. Would you be up to making a pull request?

**Comment by mariusschulz** [comment]
Sure, I'll give it a spin.

**Comment by mariusschulz** [comment]
Here's the commit which does the change for JavaScript:

https://github.com/mariusschulz/highlight.js/commit/10971e2b1096bf0eafd2a2ba4a5dbc233d1b9e1b

How do you want to deal with other languages, shouldn't they all behave similarly? In that case, would you like me to update them all, at least those that define params within parentheses?

**Comment by isagalaev** [comment]
Oh, if you could go to the trouble of updating the other languages that would be awesome! Thanks!

**Comment by mariusschulz** [comment]
For now, I've updated the following languages:

- JavaScript
- TypeScript
- C#

I've also added a simple markup test for TypeScript, which didn't exist before.

To avoid creating empty `<span class="hljs-params">` tags for parameterless functions, I've added a negative lookahead to the regular expression detecting parameters. The pattern only matches if the opening parenthesis isn't immediately followed by a closing one (with optional whitespace in between):

```js
{
  begin: /\((?!\s*\))/
}
```

I also wanted to update the rules for Swift, however that didn't work the same way as the other languages. I'm new to your codebase, so I'm not sure how to solve that. I'll leave it at this for now.
